src/gradient_store.rs

```diff
@@ -0,0 +1,150 @@
+use std::collections::BTreeMap;
+
+use imgref;
+use rgb;
+
+use crate::{Color, ErrorKind, ImageFlags, ImageId, ImageInfo, ImageSource, Renderer, image::ImageStore, paint::MultiStopGradient};
+
+/// GradientStore holds image ids for multi-stop gradients. The actual image/textures
+/// are contained by the Canvas's ImageStore.
+//
+// If many gradients are used in a frame, we could combine them into a single texture
+// and update the texture immediately prior to giving the renderer the command list.
+pub(crate) struct GradientStore {
+    this_frame: BTreeMap<MultiStopGradient, ImageId>,
```

Comment:
I’m just curious, is there a specific reason to prefer BTreeMap over HashMap here? Do you need the ordering of keys?

Reply:
No, I used `BTreeMap` because it was easier to do `Ord` for the floating point values in `MultiStopGradient` than to do `Hash`. I bet if you have fewer than 5 multistop gradients in a frame that we'd actually be faster using a flat array and doing a linear search, but this was easier to implement and is unlikely to be very hot.

Reply:
Makes perfect sense - thanks :)